### PR TITLE
[FEAT] ALL INVITATIONS ENDPOINT

### PR DESCRIPTION
**Pull request by Olatomiw**
<h1> How should this be manually tested </h1>

<h1>Accept user invite link</h1>
- This feature can be tested on Postman manually by copying and pasting the invitation link in the request body of 

```
"api/v1/organisation/invite" 
```
```
{"invitationLink": "http://invite/accept?token=a254d64f-50f0-4b28-bb70-d85b1c257763"} 
```

- Use the create invite link endpoint to generate invitation link
- This is a protected endpoint. So you also need your token before you can access it

<h1>Send Invite Endpoint</h1>

```
api/v1/organisations/send-invite
```
<h3>Reqeust Body</h3>

```
{ "email":["Johndoe14@gmail.com", "Jamesokanlawon@gmail.com"],
    "organisation_id" : "77093b16-1c02-436f-8b9b-39b3e681eae1"
}
```
This endpoint can create multiple invite links and send to the provided email. Using Java Mail sender

- it generates unique invite link for each email address provided

N.b: All invitation links will expire 2 days after it was created
<h1>Checklist of what I did</h1>

- [x] Validate the the invitation Link. i.e check if the organisation exist and the invitation link has not expired yet
- [x] I also check if user doesn't already belong to the organization. To avoid duplicate
- [x]  Accept the appropriate user into the desired organization
- [x] Unit testing to check if the link has expired, the link is malformed and finally accepts the user into the organisation

## Reference of Issue
[https://github.com/hngprojects/hng_boilerplate_java_web/issues/4#issue-2416092868](ok)

### CODE REVIEW

File: src/main/resources/db/migration/V17__create_invitation_table.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+-- V2__create_invitation_table.sql
+CREATE TABLE invitation_table (
+    id VARCHAR(255) PRIMARY KEY,
+    token VARCHAR(255) NOT NULL UNIQUE,
+    user_email VARCHAR(255) NOT NULL,
+    status VARCHAR(20) NOT NULL,
+    expires_at TIMESTAMP NOT NULL,
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL,
```
Comment (lurldgbodex) [inline src/main/resources/db/migration/V17__create_invitation_table.sql:8]: check the repo and update the version no accordingly.

**Review by lurldgbodex — CHANGES_REQUESTED** [review]
makes changes as requested

File: src/main/resources/db/migration/V17__create_invitation_table.sql (hunk 1 above)
Comment (Olatomiw) [reply src/main/resources/db/migration/V17__create_invitation_table.sql]: Done

File: src/main/resources/db/migration/V23__create_email_template_table.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CREATE TABLE email_templates (
+ id VARCHAR(36) PRIMARY KEY,
+ title VARCHAR(255) NOT NULL UNIQUE,
+ template TEXT NOT NULL,
+ status VARCHAR(255),
```
Comment (lurldgbodex) [inline src/main/resources/db/migration/V23__create_email_template_table.sql:5]: there is a V23 of the migration file

**Review by lurldgbodex — CHANGES_REQUESTED** [review]
put all your migrations in a single file and update version number

File: src/main/resources/db/migration/V23__create_email_template_table.sql (hunk 2 above)
Comment (Olatomiw) [reply src/main/resources/db/migration/V23__create_email_template_table.sql:5]: I didn't touch any other version of the migration file.
I just did my own which is v25.

**Pull request closed without merging**